### DISCUSSION TITLE
Very bad performance while developing roslyn - small projects work just fine

### DISCUSSION THREAD

**Discussion by bernd5** [dotnet/roslyn#51211] [General] [answered]
While developing roslyn my Visual studio (16.9.0 Preview 4) is often not responding - is this only the case on my machine? What are your experiences?
The solution seems to be just to big - how do you work with it? Are there any tricks to get a responsive IDE?

**Comment by CyrusNajmabadi** [comment, marked as answer]
Can you follow the instructions here to report performance issues to us: https://github.com/dotnet/roslyn/blob/master/docs/wiki/Reporting-Visual-Studio-crashes-and-performance-issues.md

After you report an issue, link to it from here and I can have someone look at it and see if it is known. 

One thing I do to help with perf is open roslyn.dln and then go make a tea. By the time I'm fine, vs had loaded everything and settled down.

**Reply by CyrusNajmabadi** [reply]
I also turn off code lens.

**Reply by bernd5** [reply]
Thanks, drinking / making a tea really helps 😀👍

**Reply by jasonmalinowski** [reply]
And if it makes you feel any better, we use the Roslyn solution internally for measuring performance improvements...

**Comment by acaly** [comment]
I also found my VS to be extremely slow when I develop a solution with internal source generator (a project inside the solution as source generator for other projects). My source generator scans every source file and every class, but I guess that's what a source generator should do.

I found it useful to clean that source generator project manually after each build. You probably cannot do this if there is an external source generator or analyzer library referenced by dll.

**Reply by jasonmalinowski** [reply]
If you're having performance issues with a generator you've written, feel free to start a separate discussion and we can help investigate what's up. https://github.com/dotnet/roslyn/blob/master/docs/wiki/Reporting-Visual-Studio-crashes-and-performance-issues.md#performance-issues gives some explanation of how to capture a trace and if you create a discussion item linking to the feedback ticket we route it more quickly.

**Comment by huoyaoyuan** [comment]
Full solution code analysis with solution error explorer can slow performance very much on huge solutions.
